File: 0125-valid-palindrome/0125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    let polishedString = s.replace(/\W+|_/g, "").toLowerCase()
    let reversedString = polishedString.split("").reverse().join("")

    if (polishedString != reversedString) {
        return false
    }
    return true
};

File: 0125-valid-palindrome/0125-valid-palindrome.cs
using System.Text.RegularExpressions;

public class Solution {
    public bool IsPalindrome(string s) {
        string polishedString = Regex.Replace(s, "[^a-zA-Z0-9]", "").ToLower();
        
        if(polishedString == "")
        {
            return true;
        }
        
        char[] charArray = polishedString.ToCharArray();
        Array.Reverse(charArray);
        string reverseString = new string(charArray);
        
        return polishedString == reverseString ? true : false;
    }
}